editor/editor_run.cpp

```diff
@@ -171,15 +171,50 @@ Error EditorRun::run(const String &p_scene, const String &p_custom_args, const L
 		args.push_back(p_scene);
 	}
 
+	String exec = OS::get_singleton()->get_executable_path();
+
 	if (p_custom_args != "") {
-		Vector<String> cargs = p_custom_args.split(" ", false);
-		for (int i = 0; i < cargs.size(); i++) {
-			args.push_back(cargs[i].replace(" ", "%20"));
+		// Allow the user to specify a command to run, similar to Steam's launch options.
+		// In this case, Godot will no longer be run directly; it's up to the underlying command
+		// to run it. For instance, this can be used on Linux to force a running project
+		// to use Optimus using `prime-run` or similar.
+		// Example: `prime-run %command% --time-scale 0.5`
+		const int placeholder_pos = p_custom_args.find("%command%");
+
+		Vector<String> custom_args;
+
+		if (placeholder_pos != -1) {
+			// Prepend executable-specific custom arguments.
+			// If nothing is placed before `%command%`, behave as if no placeholder was specified.
+			Vector<String> exec_args = p_custom_args.substr(0, placeholder_pos).split(" ", false);
+			if (exec_args.size() >= 1) {
+				exec = exec_args[0];
+				exec_args.remove(0);
+
+				// Append the Godot executable name before we append executable arguments
+				// (since the order is reversed when using `push_front()`).
+				args.push_front(OS::get_singleton()->get_executable_path());
+			}
+
+			for (int i = exec_args.size() - 1; i >= 0; i--) {
+				// Iterate backwards as we're pushing items in the reverse order.
+				args.push_front(exec_args[i].replace(" ", "%20"));
```

Comment:
Why does it use `push_front` instead of `push_back`?

Reply:
`args` already has some Godot-specific arguments appended at this point, so the executable arguments need to be *preprended*. Otherwise, the executable won't use them and Godot will try to interpret them instead.

I agree it can be a bit confusing, but I didn't feel like refactoring the whole thing just for this :slightly_smiling_face: